core/src/main/scala/org/apache/spark/partial/ApproximateActionListener.scala

```diff
@@ -34,11 +34,13 @@ private[spark] class ApproximateActionListener[T, U, R](
     rdd: RDD[T],
     func: (TaskContext, Iterator[T]) => U,
     evaluator: ApproximateEvaluator[U, R],
-    timeout: Long)
+    timeout: Long = 1000*60*60*24*30*12,
```

Comment:
You won't be able to do this because it changes the binary API. This is also mixing up two different semantics: wait for an amount of time, or wait for an amount of completion. I get the use case for both, but if this is intended to deal with skew, I think it's probably not the right solution in general. You need to deal with the skew more directly. It's not clear that ignoring 1 partition is the right thing to do, especially when it contains a lot of the data.